### PR TITLE
Stop caching fragments on SearchFragment

### LINKED ISSUE #590: Memory leak by tab logic
## Overview (Required)
- This leak is not only sessions 😭 
- https://github.com/DroidKaigi/conference-app-2018/pull/577#issuecomment-362874431

image

### PR DESCRIPTION
**Pull request by ogapants**
## Issue
- close #590

## Overview (Required)
- Stop caching fragments on SearchFragment. like this https://github.com/DroidKaigi/conference-app-2018/pull/493/files#diff-8fa8830b5e88e5e809ec4e5ea4be2e6cR195
- Actually, I don't know whether this problem has definitely been fixed.
But this has not occured now... 🤔

### CODE REVIEW

**Comment by takahirom** [comment]
🙇

File: app/src/main/java/io/github/droidkaigi/confsched2018/presentation/search/SearchFragment.kt [hunk 1]
```diff
@@ -236,10 +236,21 @@ class SearchBeforeViewPagerAdapter(
         }
     }
 
-    enum class Tab(@StringRes val title: Int, val fragment: Fragment) {
-        Session(R.string.search_before_tab_session, SearchSessionsFragment.newInstance()),
-        Topic(R.string.search_before_tab_topic, SearchTopicsFragment.newInstance()),
-        Speakers(R.string.search_before_tab_speaker, SearchSpeakersFragment.newInstance());
+    enum class Tab(@StringRes val title: Int) {
+        Session(R.string.search_before_tab_session) {
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/presentation/search/SearchFragment.kt]: This is originally, if you have time, can you use capital letters? 🙇 
https://android.github.io/kotlin-guides/style.html#enum-classes

Comment (ogapants) [reply app/src/main/java/io/github/droidkaigi/confsched2018/presentation/search/SearchFragment.kt]: Okay! Please wait a little 🙏

**Comment by ogapants** [comment]
I also fixed other lowercase enums 🤓

**Comment by ogapants** [comment]
Sorry! Just nothing! Forget it!

**Comment by takahirom** [comment]
Thanks!🎉

**Pull request merged**